widgets/Firestore-batchsink.json

```diff
@@ -47,19 +55,43 @@
       "properties": [
         {
           "widget-type": "textbox",
-          "label": "Service Account File Path",
-          "name": "serviceFilePath",
+          "label": "Project ID",
+          "name": "project",
           "widget-attributes" : {
             "default": "auto-detect"
           }
         },
+        {
+          "name": "serviceAccountType",
+          "label": "Service Account Type",
+          "widget-type": "radio-group",
+          "widget-attributes": {
+            "layout": "inline",
+            "default": "filePath",
+            "options": [
+              {
+                "id": "filePath",
+                "label": "File Path"
+              },
+              {
+                "id": "JSON",
+                "label": "JSON"
+              }
+            ]
+          }
+        },
         {
           "widget-type": "textbox",
-          "label": "Project ID",
-          "name": "project",
+          "label": "Service Account File Path",
+          "name": "serviceFilePath",
           "widget-attributes" : {
             "default": "auto-detect"
           }
+        },
+        {
+          "widget-type": "password",
```

Comment:
we use widget-type `text` for json in other plugins.

Reply:
But, that means that if there is a Service Account JSON, it would always be visible in plain text instead of password like field. Does this still need to be in text?

Reply:
yeah I am not entirely sure about the reason, it has been historically like that so to be consistent I suggested that.

Reply:
Generally customers use [secure macro](https://cdap.atlassian.net/wiki/spaces/DOCS/pages/1188036697/Macros+and+macro+functions#Secure-Function) for providing service account json in the plugin.

Reply:
Got it. Thanks for the explanation!